src/target/linux/memory.rs

```diff
@@ -0,0 +1,36 @@
+//! Utility functions to work with memory.
+
+use crate::target::MemoryMap;
+use std::cmp::Ordering;
+
+lazy_static::lazy_static! {
+    /// Memory page size from system configuration.
+    pub(crate) static ref PAGE_SIZE: usize = unsafe { libc::sysconf(libc::_SC_PAGESIZE) as usize };
+}
+
+/// Abstract memory operation (reading or writing).
+pub trait MemoryOp {
+    /// Returns a remote address at which this memory operation will be applied to.
+    fn remote_base(&self) -> usize;
+}
+
+/// Splits memory operations to those that read from read protected memory and those that do not.
+pub fn split_protected<'a, M: MemoryOp + Clone>(
```

Comment:
This is also necessary for write protected memory.

Reply:
Good catch, thanks - I made it generic but haven't changed the comments and variable names for some reason :sweat_smile:

Reply:
It should check for `is_writable` in the write case.